dna/data.py

```diff
@@ -361,3 +379,63 @@ def _iter(self):
 
     def __len__(self):
         return len(self._group_batches)
+
+class RNNDataset(Dataset):
+    def __init__(self, data: dict, features_key: str, target_key: str, task_type: str, device: str,
+                 prim_to_enc: dict):
+        super(RNNDataset, self).__init__(data, features_key, target_key, task_type, device)
+        self.pipeline_key = 'pipeline'
+        self.primitive_to_enc = prim_to_enc
+
+    def __getitem__(self, index):
+        (x, y) = super().__getitem__(index)
+        item = self.data[index]
+        pipeline = item[self.pipeline_key]
+        encoded_pipeline = self.encode_pipeline(pipeline)
+        return (encoded_pipeline, x, y)
+
+    def encode_pipeline(self, pipeline):
+        # Create a tensor of encoded primitives and a list of the indices of each primitive's inputs
+        encoding = []
+        for primitive in pipeline['steps']:
+            primitive_name = primitive['name']
+            encoded_primitive = self.primitive_to_enc[primitive_name]
+            encoding.append(encoded_primitive)
+
+        encoding = torch.tensor(encoding, dtype=torch.float32)
+
+        if "cuda" in self.device:
+            encoding = encoding.cuda()
+
+        return encoding
+
+class RNNDataLoader(GroupDataLoader):
```

Comment:
Why do we need a new dataset and dataloader for the RNN? Can we use the existing ones with different arguments?

Reply:
After some discussion with @erikhuck, It may be possible to make a general dataset and dataloader class, but the solution would require some refactoring and time. We will spend our time elsewhere.